src/main/groovy/com/etendoerp/legacy/LegacyScriptLoader.groovy

```diff
@@ -102,7 +102,8 @@ class LegacyScriptLoader {
         if (project.file('modules').exists() && project.file('modules').isDirectory()) {
             project.file('modules').eachDir {
                 project.sourceSets.main.java.srcDirs += it.toString() + "/src"
-                if (project.file(it.toString() + '/etendo-resources').exists()) {
```

Comment:
There are some improvements that can be made to make the code more readable and maintainable:

Extract the "etendo-resources" directory into a variable to avoid unnecessary code repetition..

Use methods of the File class instead of converting to string to check for the existence of directories.

Use the operator (?:) to simplify logic.

`def resourcesDir = project.sourceSets.main.resources.srcDirs.find { it.name == 'etendo-resources' }
def etendoResourcesDir = new File(project.projectDir, 'etendo-resources')

if (!resourcesDir && etendoResourcesDir.exists()) {
    project.sourceSets.main.resources.srcDirs += etendoResourcesDir
}
`